node.go

```diff
@@ -310,6 +310,10 @@ func (nw *NodeWalker) resumeAt(node *Node, entering bool) (*Node, bool) {
 	return nw.next()
 }
 
+func (ast *Node) String() string {
+	return dumpString(ast)
+}
```

Comment:
Do I understand correctly that this would dump the entire AST of the document recursively? Wouldn't that be a very large output?

I'm not sure if that's a great fit for `fmt.Stringer` output. Usually it's meant to be short descriptions that can fit into `fmt.Errorf("something went wrong with %v in FooBar: %v", htmlRenderer, err)` messages, etc.

Do you want this functionality for debugging?